catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/services/ElasticsearchServiceTest.scala

```diff
@@ -1,121 +1,182 @@
 package uk.ac.wellcome.platform.api.services
 
 import com.sksamuel.elastic4s.http.search.SearchHit
+import org.scalatest.compatible.Assertion
 import org.scalatest.concurrent.ScalaFutures
 import org.scalatest.{FunSpec, Matchers}
 import uk.ac.wellcome.models.IdentifiedWork
 import uk.ac.wellcome.platform.api.WorksUtil
 import uk.ac.wellcome.models.WorksIncludes
 import uk.ac.wellcome.display.models.DisplayWork
-import uk.ac.wellcome.elasticsearch.test.utils.IndexedElasticSearchLocal
+import uk.ac.wellcome.elasticsearch.test.fixtures.ElasticsearchFixtures
+import uk.ac.wellcome.test.fixtures.TestWith
 import uk.ac.wellcome.utils.JsonUtil._
 
 class ElasticsearchServiceTest
     extends FunSpec
-    with IndexedElasticSearchLocal
     with Matchers
     with ScalaFutures
+    with ElasticsearchFixtures
     with WorksUtil {
 
-  val indexName = "works"
   val itemType = "work"
 
-  val searchService =
-    new ElasticSearchService(indexName, itemType, elasticClient)
+  private def withElasticSearchService(indexName: String)(
+    testWith: TestWith[ElasticSearchService, Assertion]) = {
 
-  it("should sort results from Elasticsearch in the correct order") {
-    val work1 = workWith(
-      canonicalId = "000Z",
-      title = "Amid an Aegean"
-    )
-    val work2 = workWith(
-      canonicalId = "000Y",
-      title = "Before a Bengal"
-    )
-    val work3 = workWith(
-      canonicalId = "000X",
-      title = "Circling a Cheetah"
+    val searchService = new ElasticSearchService(
+      defaultIndex = indexName,
+      documentType = itemType,
+      elasticClient = elasticClient
     )
 
-    insertIntoElasticSearch(work1, work2, work3)
-
-    assertSliceIsCorrect(
-      limit = 3,
-      from = 0,
-      expectedWorks = List(work3, work2, work1).map { DisplayWork(_) }
-    )
+    testWith(searchService)
+  }
 
-    // TODO: canonicalID is the only user-defined field that we can sort on.
-    // When we have other fields we can sort on, we should extend this test
-    // for different sort orders.
+  it("should sort results from Elasticsearch in the correct order") {
+    withLocalElasticsearchIndex(itemType = itemType) { indexName =>
```

Comment:
I noticed some test instability using this version of the method, vs the async `withLocalElasticsearchIndexAsync` version. When re-factoring the `common` tests.

Reply:
Okay, maybe when I come back and do the itemType stuff, I’ll get rid of all the non-async methods at the same time.

Reply:
Cheers, Alex.